WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -400,6 +420,21 @@ private async Task InitialiseViewModelAsync()
 		if (await BuildTransactionAsync(BuildTransactionReason.Initialization) is { } initialTransaction)
 		{
 			UpdateTransaction(CurrentTransactionSummary, initialTransaction);
+
+			if (!_info.IsAutomaticSelectionEnabled)
+			{
+				var coinSelectionDialogResult = await NavigateDialogAsync(new SelectCoinsDialogViewModel(_walletViewModel, _info, _balanceChanged, Transaction?.SpentCoins));
+
+				if (coinSelectionDialogResult.Kind == DialogResultKind.Normal && coinSelectionDialogResult.Result is { })
+				{
+					_info.Coins = coinSelectionDialogResult.Result;
+					await BuildAndUpdateAsync(BuildTransactionReason.PocketChanged, addToUndoHistory: false);
+				}
+				else
+				{
+					Navigate().Back();
+				}
+			}
```

Comment:
Are you sure this is the right place for this? it seems to me the transaction preview shouldn't handle what happens *before* it, but rather inside it.

Anyways, we were discussing about removing the toggle from the first dialog and putting it in the transaction preview. Would that be feasible?

Reply:
This is @soosr's code. I didn't modify the flow myself.